### ISSUE TITLE
Reporting of characters in input help mode is unreliable if the current input language is not english

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#14793]
### Steps to reproduce:
1. Add another (preferably - non-latin) input language to Windows (in my case it is bulgarian), so you have two - english and the additional one.
2. Switch to the newly added input language.
3. Preferably, switch to a multilingual speech synthesizer (e.g. eSpeak), which can read the letters from both english and the additional input language added in step 1.
4. Enable input help mode in NVDA.
5. Press each key with a letter/character from the main part of the keyboard.
6. Observe what will NVDA report for each pressed key.

### Actual behavior:
In most cases, NVDA reports the letters from the english input language. Exceptions are the keys where for the english input language/layout there are symbols/punctuation characters instead of letters - there NVDA reports the corresponding characters from the additional input language.

### Expected behavior:
NVDA reports the characters (symbols/punctuation/letters) for the currently active input language (e.g. in my case - bulgarian).

### NVDA logs, crash dumps and other attachments:
[NVDA_InputHelp.log](https://github.com/nvaccess/nvda/files/11159653/NVDA_InputHelp.log)

#### NVDA installed/portable/running from source:
Installed.

#### NVDA version:
NVDA version alpha-28022,5f340be5

#### Windows version:
Windows 11 Pro version 22H2 (build 22621.1413), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".

#### Name and version of other software in use when reproducing the issue:
N/A

#### Other information about your system:
N/A

#### Does the issue still occur after restarting your computer?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
NVDA 2023.1 - the behavior is the same.

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes, it is.

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A.

**Comment by k-kolev1985** [comment]
@gerald-hartig 
For the bulgarian keyboard layouts. The one that is named "Bulgarian" or the one named "Bulgarian (Typewriter)" will do. Maybe the phonetic one as well, but I don't use that and I'm not familiar with it.
As for which characters are affected. It depends on the used bulgarian layout. As I wrote in the bug's description, each bulgarian character which is on the same key as a latin character is affected. Not affected are the ones that are on the same key as a punctuation character from the english layout.
Hope this helps.

**Comment by CyrilleB79** [comment]
> [@gerald-hartig](https://github.com/gerald-hartig) For the bulgarian keyboard layouts. The one that is named "Bulgarian" or the one named "Bulgarian (Typewriter)" will do. Maybe the phonetic one as well, but I don't use that and I'm not familiar with it. As for which characters are affected. It depends on the used bulgarian layout. As I wrote in the bug's description, each bulgarian character which is on the same key as a latin character is affected. Not affected are the ones that are on the same key as a punctuation character from the english layout. Hope this helps.

@k-kolev1985, please provide a specific example, with a specific keyboard layout.
For example with the one that is named "Bulgarian", provide us with:
* the character at the "Q" position of the English layout, and confirm that "Q" is reported
* the character at the position of "," (comma) of English layout, and confirm the Bulgarian character being reported.

Thanks

**Comment by k-kolev1985** [comment]
Alright, here it goes. I'll use the "Bulgarian" keyboard layout and the "English US" one.

English layout	Bulgarian layout	What NVDA reports in input help mode with layout set to “Bulgarian”
Q	,	,
W	W	W
E	E	E
R	R	R
T	T	T
Y	Y	Y
I	I	I
O	O	O
P	P	P
[	Ц	Ц
]	;	;
A	A	A
S	S	S
D	D	D
F	F	F
G	G	G
H	H	H
J	J	J
K	K	K
L	L	L
;	М	М
‘	Ч	Ч
\	„	„
\	Ѝ	Ѝ
Z	Z	Z
X	X	X
C	C	C
V	V	V
B	B	B
N	N	N
M	M	M
,	Р	Р
.	Л	Q
/	Б	Б

Hope this helps.

**Comment by CyrilleB79** [comment]
Sorry @k-kolev1985 it's not yet clear to me.
For example to type "български", how do you type letter "г" given this does not appear in the Bulgarian layout you have provided above?

With this specific example, could you answer:
* Where is "г" on the Bulgarian layout?
* What is the corresponding letter (if any) or symbol on the English layout?
* If you type "г" while NVDA is in input help mode, what is reported?

**Comment by k-kolev1985** [comment]
@CyrilleB79 
OK, I'll try to answer your questions to the best of my ability.
- The letter "Г" is on the middle of the second row, between the letters "ж" and "т".
- The corresponding character on the english layout is "h".
- When I type "г" with input help on, NVDA reports "h".
Hope this helps.

**Comment by CyrilleB79** [comment]
OK. Seem clearer to me. The confusion came from the fact that you had not put all Bulgarian letters (such as "г") in the description of Bulgarian layout in https://github.com/nvaccess/nvda/issues/14793#issuecomment-2812297762.

So if I summarize, in input help mode:
* you type the "г" key, i.e. the letter at the position of "H" on the English keyboard
* you hear "H"
* you expect "г"

**Comment by k-kolev1985** [comment]
@CyrilleB79 
Yes, you've understood the actual and expected results correctly.
And I apologize, because I messed up the presentation with the 3 columns with the alphabet. The second column should have always contained the bulgarian letter, but in most cases it contained the english one instead.
